### ISSUE TITLE
Currency: support symbol-alt-narrow

### ISSUE THREAD

**Issue by Renkas** [globalizejs/globalize#479]
CLDR exposes also symbol-alt-narrow for some currencies (since CLDR 24: http://cldr.unicode.org/index/downloads/cldr-24)

Example in 'et' locale Globalize.currencyFormatter( "PLN" )( 1 ) - outputs:
1,00 PLN

With symbol-alt-narrow support the output should be:
1,00 zł

**Comment by rxaviers** [comment]
CLDR indeed has the data. But, it doesn't specify how to use it. This same topic has been discussed in https://github.com/tc39/ecma402/issues/4. So, the first step is to define how this could be supported.

I recommend researching CLDR tickets (e.g., http://unicode.org/cldr/trac/search?changeset=on&milestone=on&ticket=on&wiki=on&q=symbol+alt&page=2&noquickjump=1) and its documentation (http://www.unicode.org/reports/tr35/tr35-numbers.html#Currencies) to find substance to elaborate a suggestion. Perhaps, a [variant subtag][] could be specified in the locale to define the preference for the alt symbol. Perhaps, an additional option on the currencyFormatter could be specified. Either way, I suggest that whatever is considered here should be aligned with CLDR.

[variant subtag]: http://www.unicode.org/reports/tr35/#unicode_variant_subtag

**Comment by Renkas** [comment]
For my purpose an additional option for the currencyFormatter() would be sufficient.

Quote from the change log:
> The narrow forms are intended to be as narrow as possible, however, there may be no good narrow term for some units in some languages. As always, narrow forms are not necessary unique; they should only be used where the context is sufficiently clear (eg, it is a length).

I think that adding an option named 'narrow' to currencyFormatter() would then do exactly what it's meant to. As the usage is context sensitive I don't think that would be the perfect place. It wouldn't make sense to make it some global option.

**Comment by rxaviers** [comment]
A `{form: "narrow"}` option sounds good to me. Basically, it should try `symbol-alt-narrow || symbol` given narrow is not available for all currencies.

Given the change log makes reference for the narrow form. But, http://www.unicode.org/reports/tr35/tr35-numbers.html#Currencies does not. I believe we should request it to be inlcuded by filing a ticket on CLDR (http://unicode.org/cldr/trac/) as a preliminary step. If you agree, please do so and add a new comment here with the ticket number. I will help bringing this to focus on the CLDR committee.

**Comment by Renkas** [comment]
There seems to be already a ticket: http://unicode.org/cldr/trac/ticket/8461

**Comment by rxaviers** [comment]
It's targeted for CLDR 28, so we could give the implementation a try. Basically, what needs update is:
- https://github.com/jquery/globalize/blob/master/src/currency.js
- https://github.com/jquery/globalize/blob/master/src/currency/symbol-properties.js#L18-L22

Feel free to submit a Pull Request. :)

**Comment by Renkas** [comment]
I'll better leave this to someone who knows what he is doing :)

Anyway thanks in advance - can't wait to use this.

**Comment by zolitch** [comment]
Looks like this has now been added to the spec: http://unicode.org/cldr/trac/ticket/8405#comment:7 

Is there any movement here or is someone working on an implementation? It would be great to get this feature!

**Comment by rxaviers** [comment]
Hey @zolitch, no one seems to be working on it, but I believe it's a quick change. Would you be interested in implementing it? Just let me know if you have any questions? Thanks

**Comment by zolitch** [comment]
I can't right now but can do if I get some time over the next week or. Provided a workaround on the project I'm working on right now.

**Comment by sreekanthjnayar** [comment]
Has this been integrated into the current version? Would be a useful feature to have, any ETA on this.

**Comment by rxaviers** [comment]
Hi @sreekanthjnayar thanks for your interest. The implementation is close to complete, it misses some review fixes #631 (which stalled). Would you be interested in moving forward with this work (see #631)? Just let me know if you have any questions? Thanks

**Comment by rxaviers** [comment]
This change is technically ready to merge. It's waiting on [Foundation matters](https://github.com/globalizejs/globalize/pull/631#issuecomment-304721783).

**Comment by kborchers** [comment]
This is good to go as you can see in #738 which shows all checks passing, including @gethinwebster's signature which I have also verified is in the database.

**Comment by rxaviers** [comment]
Yeap. Thanks... @gethinwebster has re-signed it.

**Comment by mihaellenic** [comment]
When could this be merged into master? It looks like there are no blockers.
I would like to use this feature very much. :)
